### PR TITLE
Attempt at solving #112 by providing key/value hash

### PR DESCRIPTION
**Pull request by oysteing**

### CODE REVIEW

File: spec/classes/jira_config_spec.rb [hunk 1]
```diff
@@ -351,6 +351,20 @@
...
+          let(:params) {{
+            :version  => '6.3.4a',
+            :javahome => '/opt/java',
+            :jiraConfigProperties => {
+              'ops.bar.group.size.opsbar-transitions' => '4',
+            }
+          }}
+          it { should contain_file('/home/jira/jira-config.properties')
+            .with_content(/jira.websudo.is.disabled = false/) }
+            .with_content(/ops.bar.group.size.opsbar-transitions = 4/) }
```
Comment (rnelson0) [inline spec/classes/jira_config_spec.rb]: I don't think you need these trailing `}`'s

Comment (igalic) [inline spec/classes/jira_config_spec.rb]: yeah, not like this is a sensible language like lisp :P

**Comment by oysteing** [comment]
Embarassing. I blame it on the Lisp gnome sneaking those braces in.

What do you think of the request in general?
* Hash solution vs other solutions?
* The new parameter?
* Camel case? There is a mix of camel case and snake case, I chose according to my preference, but feel free to instruct.
* Merging the enable_secure_admin_sessions setting into the hash?

**Comment by igalic** [comment]
camelCase is discouraged in the [style guide](https://docs.puppetlabs.com/guides/style_guide.html#variable-format)

**Comment by igalic** [comment]
if we remove `enable_secure_admin_sessions` we're breaking compatibility, if we merge we have to consider both names

**Comment by oysteing** [comment]
I've changed to snake_case. If there are no more comments, I consider myself done.

**Comment by igalic** [comment]
@oysteing only one more thing: could you please [squash](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits) this down to one commit?
****
n.b.: if you use the [github flow](https://guides.github.com/introduction/flow/index.html) work-flow this might be easier or less confusing
see also: https://github.com/voxpupuli/puppet-jira/pull/99#issuecomment-152528108 for a step-by-step how-to.

**Pull request merged**